pkg/activator/certificate/cache.go

```diff
@@ -101,9 +105,32 @@ func (cr *CertCache) updateCache(secret *corev1.Secret) {
 	}
 	pool.AddCert(ca)
 
-	cr.TLSConf.RootCAs = pool
-	cr.TLSConf.ServerName = certificates.LegacyFakeDnsName
-	cr.TLSConf.MinVersion = tls.VersionTLS12
+	cr.ClientTLSConf.RootCAs = pool
+	cr.ClientTLSConf.ServerName = certificates.LegacyFakeDnsName
+	cr.ClientTLSConf.MinVersion = tls.VersionTLS12
+	cr.ClientTLSConf.Certificates = []tls.Certificate{cert}
+
+	cr.ServerTLSConf.MinVersion = tls.VersionTLS12
+	cr.ServerTLSConf.GetCertificate = cr.GetCertificate
+
+	switch cr.trust {
+	case netcfg.TrustIdentity, netcfg.TrustMutual:
+		cr.ServerTLSConf.ClientAuth = tls.RequireAndVerifyClientCert
+		cr.ServerTLSConf.ClientCAs = pool
+		cr.ServerTLSConf.VerifyConnection = func(cs tls.ConnectionState) error {
+			for _, match := range cs.PeerCertificates[0].DNSNames {
+				if match == "kn-routing-0" { // routingId not yet supported
```

Comment:
That comment is a bit misleading when the context is unknown. It is clear because I know the history. Could you explain that `kn-routing-0` currently is the default sent and other values are not yet expected because it is not yet implemented?

Reply:
Shouldn't these be constants or string templates from `networking`. Seems a bit of as a string here.